types/src/v0/impls/stake_table.rs

```diff
@@ -974,6 +981,165 @@ impl StakeTableFetcher {
+
+                tracing::debug!(tx_hash = ?init_log.transaction_hash, "Found token `Initialized` event");
+                init_log
+            },
+            Err(err) => {
+                tracing::warn!(
+                    "RPC returned error {err:?}. will fallback to scanning over fixed block range"
+                );
+                self.scan_token_contract_initialized_event_log(stake_table_init_block, token)
+                    .await?
+            },
+        };
+
+        // Get the transaction that emitted the Initialized event
+        let init_tx = provider
+            .get_transaction_receipt(
+                init_log
+                    .transaction_hash
+                    .context(format!("transaction hash not found. init_log={init_log:?}"))?,
+            )
+            .await?
+            .context(format!(
+                "Failed to get transaction for Initialized event. hash={:?}",
+                init_log.transaction_hash
+            ))?;
+
+        let mint_transfer = init_tx
+            .decoded_log::<EspToken::Transfer>()
+            .context(format!(
+                "Failed to decode Transfer log. init_tx={:?}",
+                init_tx
+            ))?;
+
+        tracing::debug!("mint transfer event ={mint_transfer:?}");
+        ensure!(
+            mint_transfer.from == Address::ZERO,
+            "First transfer should be a mint from the zero address"
+        );
+
+        let initial_supply = mint_transfer.value;
+
+        tracing::info!("Initial token amount: {} ESP", format_ether(initial_supply));
+
+        let reward = ((initial_supply * U256::from(INFLATION_RATE)) / U256::from(BLOCKS_PER_YEAR))
+            .checked_div(U256::from(COMMISSION_BASIS_POINTS))
+            .ok_or_else(|| anyhow::anyhow!("COMMISSION_BASIS_POINTS is zero"))?;
+
+        Ok(RewardAmount(reward))
+    }
+
+    pub async fn scan_token_contract_initialized_event_log(
+        &self,
+        stake_table_init_block: u64,
+        token: EspTokenInstance<(), L1Provider>,
+    ) -> anyhow::Result<Log> {
+        let max_events_range = self.l1_client.options().l1_events_max_block_range;
+        const MAX_BLOCKS_SCANNED: u64 = 200_000;
+        let mut total_scanned = 0;
+
+        let mut from_block = stake_table_init_block.saturating_sub(max_events_range);
+
+        loop {
+            if total_scanned >= MAX_BLOCKS_SCANNED {
+                tracing::error!(
+                    total_scanned,
+                    "Exceeded maximum scan range while searching for token Initialized event"
+                );
+                return Err(anyhow::anyhow!(
+                    "No Initialized event found after scanning {} blocks",
+                    MAX_BLOCKS_SCANNED
+                ));
+            }
+            let init_logs = token
+                .Initialized_filter()
+                .from_block(from_block)
+                .to_block(stake_table_init_block)
```

Comment:
This should be something like

```suggestion
                .to_block(from_block + max_events_range)
```

shouldn't it?